docs/design.rst

```diff
@@ -0,0 +1,84 @@
+#################
```

Comment:
@itepifanio The additions that this PR brings are quite nice, but unless I'm missing something the page for Design Decisions (https://palaimon.github.io/ipyannotator/docs/design.html) is not linked anywhere from the landing pages.

![design-page](https://user-images.githubusercontent.com/5142394/186072320-76e73e7e-9ec9-4e6a-94f7-14d0614c908d.png)

Can you either point me to where I should be looking for the link, or add a link to the index/landing page (https://palaimon.github.io/ipyannotator/index.html) or to the docs navigation toolbar?

Reply:
@matthewfeickert sorry, I forgot to add the Design page into the navbar. It should be working now after #45 was merged.